src/blocks/renderers/iframe.ts

```diff
@@ -60,15 +60,29 @@ export class IFrameRenderer extends Renderer {
     title = "PixieBrix",
     height = "100%",
     width = "100%",
-    safeMode,
-  }: BlockArg): Promise<string> {
+    safeMode = false,
+  }: BlockArg<{
+    url: string;
+    title?: string;
+    height?: string;
+    width?: string;
+    safeMode?: boolean;
+  }>): Promise<SafeHTML> {
+    const parsedURL = new URL(url);
+
     if (safeMode) {
-      return `<iframe src="${url}" title="${title}" height="${height}" width="${width}" style="border:none;" allowfullscreen="false" allowpaymentrequest="false"></iframe>`;
+      return `<iframe src="${parsedURL.toString()}" title="${title}" height="${height}" width="${width}" style="border:none;" allowfullscreen="false" allowpaymentrequest="false"></iframe>` as SafeHTML;
```

Comment:
Alternatively, to convert a URL into a string:

```suggestion
      return `<iframe src="${parsedURL.href}" title="${title}" height="${height}" width="${width}" style="border:none;" allowfullscreen="false" allowpaymentrequest="false"></iframe>` as SafeHTML;
```